File: padaria-app/src/app/module/models/src/employee/marital-status.ts
import { describe } from '@module/utils/functions/enum';

export enum MaritalStatus {
  None = 0,
  Single = 1,
  Married = 2,
  Spouse = 3,
  Divorced = 4,
  Widowed = 5,
}

describe(MaritalStatus, {
  None: 'Nenhum',
  Single: 'Solteiro',
  Married: 'Casado',
  Spouse: 'Cônjuge',
  Divorced: 'Divorciado',
  Widowed: 'Viúvo',
});

File: padaria-api/src/main/java/com/padaria/model/maritalStatus/MaritalStatus.java
package com.padaria.model.maritalStatus;

public enum MaritalStatus {

    None ("Nenhum"),
    Single("Solteiro"),
    Married("Casado"),
    Spouse("Cônjuge"),
    Divorced ("Divorciado"),
    Widowed ("Viúvo");

    private final String description;

    MaritalStatus(String description) {
        this.description = description;
    }

    public String getValue() {
        return description;
    }

    @Override
    public String toString() {
        return description;
    }

}
